index.d.ts

```diff
@@ -1,45 +1,69 @@
-import { Connection, Channel } from 'amqplib/callback_api';
-import fastify, { FastifyPlugin } from 'fastify';
+import { Connection, Channel } from 'amqplib';
+import { FastifyPluginAsync } from 'fastify';
 
 declare namespace fastifyAmqp {
-  interface FastifyAmqpConnObject extends Connection {}
+  type FastifyAmqpConnObject = Connection;
 
-  interface FastifyAmqpChannelObject extends Channel {}
+  type FastifyAmqpChannelObject = Channel;
 
   interface FastifyAmqpOptions {
+    /**
+     * Full connection URL
+     */
+    url?: string;
     /**
      * Host of connection
      */
-    host: string;
+    hostname?: string;
     /**
      * Port to connect
      * @default 5672
      */
     port?: number;
     /**
      * User to connect
-     * @default guest
+     * @default 'guest'
      */
-    user?: string;
+    username?: string;
     /**
      * Password to connect
-     * @default guest
+     * @default 'guest'
+     */
+    password?: string;
```

Comment:
These new options should have a test as well.

Reply:
Those options are passed directly to `amqplib`. The plugin doesn't use them. To test all possible combinations, we'll end up rewriting `amqplib` tests, and It's out of scope.